src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1165,6 +1167,46 @@ public function testCollectDenormalizationErrorsWithConstructor(?ClassMetadataFa
         $this->assertSame($expected, $exceptionsAsArray);
     }
 
+    /**
+     * @requires PHP 8.1
+     */
+    public function testCollectDenormalizationErrorsWithEnumConstructor()
+    {
+        $serializer = new Serializer(
+            [
+                new BackedEnumNormalizer(),
+                new ObjectNormalizer(),
+            ],
+            ['json' => new JsonEncoder()]
+        );
+
+        try {
+            $serializer->deserialize('{"invalid": "get"}', DummyObjectWithEnumConstructor::class, 'json', [
+                DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+            ]);
+        } catch (\Throwable $th) {
+            $this->assertInstanceOf(PartialDenormalizationException::class, $th);
+        }
+
+        $exceptionsAsArray = array_map(function (NotNormalizableValueException $e): array {
+            return [
+                'currentType' => $e->getCurrentType(),
+                'useMessageForUser' => $e->canUseMessageForUser(),
+                'message' => $e->getMessage(),
+            ];
+        }, $th->getErrors());
+
+        $expected = [
+            [
+                'currentType' => 'array',
+                'useMessageForUser' => true,
+                'message' => 'Failed to create object because the object miss the "get" property.',
```

Comment:
The message looks misleading as the class does have a `get` property, isn't it? Shouldn't it rather say that we are not able to create an object of the expected type X from the given value X?

Reply:
indeed! 
It failed because no matching case or scalar equivalent found

Reply:
About the test mentioned above, it's not link to the BackedEnumNormalizer. Maybe I should remove it because misleading.